spec/factories/trainees.rb

```diff
@@ -111,6 +111,12 @@
       end
     end
 
+    trait :without_provider do
```

Comment:
It feels a bit wrong

I'm not sure a trainee can really ever exist without a provider